src/handlers/gestures/GestureDetector/attachHandlers.ts

```diff
@@ -0,0 +1,114 @@
+import React from 'react';
+import { GestureType, HandlerCallbacks } from '../gesture';
+import { registerHandler } from '../../handlersRegistry';
+import RNGestureHandlerModule from '../../../RNGestureHandlerModule';
+import {
+  filterConfig,
+  scheduleFlushOperations,
+} from '../../gestureHandlerCommon';
+import { ComposedGesture } from '../gestureComposition';
+import { ActionType } from '../../../ActionType';
+import { Platform } from 'react-native';
+import type RNGestureHandlerModuleWeb from '../../../RNGestureHandlerModule.web';
+import { ghQueueMicrotask } from '../../../ghQueueMicrotask';
+import { GestureConfigReference, WebEventHandler } from './types';
+import {
+  extractGestureRelations,
+  checkGestureCallbacksForWorklets,
+  ALLOWED_PROPS,
+} from './utils';
+
+interface AttachHandlersConfig {
+  preparedGesture: GestureConfigReference;
+  gestureConfig: ComposedGesture | GestureType;
+  gestures: GestureType[];
+  viewTag: number;
+  webEventHandlersRef: React.RefObject<WebEventHandler>;
+  mountedRef: React.RefObject<boolean>;
+}
+
+export function attachHandlers({
+  preparedGesture,
+  gestureConfig,
+  gestures,
+  viewTag,
+  webEventHandlersRef,
+  mountedRef,
+}: AttachHandlersConfig) {
+  gestureConfig.initialize();
```

Comment:
This was done in a weird way. The check you're mentioning was actually negated: https://github.com/software-mansion/react-native-gesture-handler/blob/81a7df0001774088f940b6ed45f2cbc9f2250de5/src/handlers/gestures/GestureDetector.tsx#L152-L156

so it would run every time but the first, then in the detector itself it was run only the first time:
https://github.com/software-mansion/react-native-gesture-handler/blob/81a7df0001774088f940b6ed45f2cbc9f2250de5/src/handlers/gestures/GestureDetector.tsx#L779-L781

Now the effect _should_ be the same, without the weird conditional gymnastics